core/service/executor.cpp

```diff
@@ -53,27 +66,81 @@ namespace executor{
         }
     }
 
+    void valueValidation(Api::Asset asset, std::vector<std::string> names){
+
+    }
+
     void transfer(const Transaction& tx){
         if(tx.has_asset()){
             // Transfer<Asset>
             auto sender = tx.senderpubkey();
             auto receiver = tx.receivepubkey();
             const auto assetName = tx.asset().name();
-            auto senderAsset    = repository::asset::find(  sender,assetName);
-            auto receiverAsset  = repository::asset::find(receiver,assetName);
+
             // **********************************************************************************
-            // * This is Transfer<Asset>'s logic.
+            // * This is Transfer<Asset>'s logic. Tax send logic
             // **********************************************************************************
+            if(tx.asset().value().find("author") != tx.asset().value().end()) {
+                const auto author = tx.asset().value().at("author").valuestring();
```

Comment:
pair `value().find(..)` & `value().at(..)` looks like performing two jobs instead one
Could it be done in one search? like:
```
auto author = tx.asset().value().find("author");
if (author != tx.asset().value().end()) {
  //...
}
```